client-spark/spark2/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -252,7 +252,9 @@ public <K, V, C> ShuffleHandle registerShuffle(int shuffleId, int numMaps, Shuff
     // get all register info according to coordinator's response
     Set<String> assignmentTags = RssSparkShuffleUtils.getAssignmentTags(sparkConf);
     ClientUtils.validateClientType(clientType);
-    assignmentTags.add(clientType);
+    if (!sparkConf.get(RssSparkConfig.RSS_TEST_MODE_ENABLE)) {
```

Comment:
Why do we need this? We should reduce similar code. If we don't have better solution, it's acceptable.